src/Microsoft.Extensions.Primitives/ChangeToken.cs

```
@@ -36,7 +36,7 @@ public static IDisposable OnChange(Func<IChangeToken> changeTokenProducer, Actio
                 // If the token changes after we take the token, then we'll process the update immediately upon
                 // registering the callback.
                 var t = changeTokenProducer();
-                changeTokenConsumer();
+                try { changeTokenConsumer(); } catch { } // Uncaught exceptions are really bad here
```

Should we really swallow the exception or could we do:

``` C#
try { changeTokenConsumer(); } 
catch { throw; }
finally { t.RegisterChangeCallback(callback, s); } 
```